src/main/java/org/broadinstitute/hellbender/utils/haplotype/HaplotypeBAMWriter.java

```diff
@@ -125,25 +103,48 @@ public void close() {
 
     /**
      * Write out a BAM representing for the haplotype caller at this site
+     * writerType (ALL_POSSIBLE_HAPLOTYPES or CALLED_HAPLOTYPES) determines inputs to writeHaplotypesAsReads
      *
      * @param haplotypes a list of all possible haplotypes at this loc
      * @param paddedReferenceLoc the span of the based reference here
      * @param bestHaplotypes a list of the best (a subset of all) haplotypes that actually went forward into genotyping
      * @param calledHaplotypes a list of the haplotypes that where actually called as non-reference
      * @param readLikelihoods a map from sample -> likelihoods for each read for each of the best haplotypes
      */
-    public abstract void writeReadsAlignedToHaplotypes(final Collection<Haplotype> haplotypes,
-                                                       final Locatable paddedReferenceLoc,
-                                                       final Collection<Haplotype> bestHaplotypes,
-                                                       final Set<Haplotype> calledHaplotypes,
-                                                       final ReadLikelihoods<Haplotype> readLikelihoods);
+    public void writeReadsAlignedToHaplotypes(final Collection<Haplotype> haplotypes,
+                                              final Locatable paddedReferenceLoc,
+                                              final Collection<Haplotype> bestHaplotypes,
+                                              final Set<Haplotype> calledHaplotypes,
+                                              final ReadLikelihoods<Haplotype> readLikelihoods) {
+
+        Utils.nonNull(haplotypes, "haplotypes cannot be null");
+        Utils.nonNull(paddedReferenceLoc, "paddedReferenceLoc cannot be null");
+        Utils.nonNull(calledHaplotypes, "calledHaplotypes cannot be null");
+        Utils.nonNull(readLikelihoods, "readLikelihoods cannot be null");
```

Comment:
We should add a null check for bestHaplotypes as well. Also, since we're now requiring calledHaplotypes to always be non-null (I think it used to depend on the writer type), lets update the javadoc for all of these args to say they must be non-null.